test/functional/cursor.test.js

```diff
@@ -405,74 +406,66 @@ describe('Cursor', function () {
           }
 
           function f() {
-            var number_of_functions = 9;
-            var finished = function () {
+            var number_of_functions = 7;
+            var finished = function (cursor) {
               number_of_functions = number_of_functions - 1;
               if (number_of_functions === 0) {
-                done();
+                cursor.close(done);
+              } else {
+                cursor.close();
               }
             };
 
             var cursor = collection.find().sort(['a', 1]);
-            test.deepEqual(['a', 1], cursor.sortValue);
-            finished();
+            test.deepEqual({ a: 1 }, cursor.sortValue);
+            finished(cursor);
 
             cursor = collection.find().sort('a', 1);
-            test.deepEqual([['a', 1]], cursor.sortValue);
-            finished();
+            test.deepEqual({ a: 1 }, cursor.sortValue);
+            finished(cursor);
 
             cursor = collection.find().sort('a', -1);
-            test.deepEqual([['a', -1]], cursor.sortValue);
-            finished();
+            test.deepEqual({ a: -1 }, cursor.sortValue);
+            finished(cursor);
 
             cursor = collection.find().sort('a', 'asc');
-            test.deepEqual([['a', 'asc']], cursor.sortValue);
-            finished();
-
-            cursor = collection.find().sort([
-              ['a', -1],
-              ['b', 1]
-            ]);
-            var entries = cursor.sortValue.entries();
```

Comment:
`.entries()` 👀 no clue what was going on here.

Reply:
It returns an iterator of tuples `[index, value]`, then each next call will iterate through the array emitting this.

Reply:
`cursor.sortValue` is typed as Sort, just the option passed into sort from the user, there's no guarantees that `.entries()` will work.

This would have broke:

```
cursor = collection.find().sort({ a: 1 });	
var entries = cursor.sortValue.entries();
```

Can we deprecate this functionality?

Reply:
Yea that makes sense, it won't work anymore, I don't think it was functionality per se just the tester knew it was still an array on the inside 👍